### PR TITLE
Template View and Model for Keynote

### PR DESCRIPTION
**Pull request by devanshgoenka97**

### CODE REVIEW

File: festflow/urls.py [hunk 1]
```diff
@@ -17,6 +17,9 @@
 
     # Event Page
     url(r'^events/$', views.events, name='events'),
+
```
Comment (ghoshbishakh) [inline festflow/urls.py]: Unnecessary spaces in line 20

File: festflow/templates/festflow/keynote_view.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+{% endif %}
+
+<h2>About the Speech</h2>
+<p>{{ keynote.description }}</p>
+<p><strong>Venue</strong>: {{ keynote.venue }}</p>
+<p><strong>Time</strong>: {{ keynote.date_time }}</p>
+
+
+</body>
+</html>
```
Comment (ghoshbishakh) [inline festflow/templates/festflow/keynote_view.html]: add a newline

File: festflow/templates/festflow/keynote.html [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>KEYNOTE SPEECHES</title>
+</head>
+<body>
+  {% for keynote in all_keynotes %}
+  <a href="{{ keynote.get_absolute_url }}">
+    {% if keynote.logo %}
```
Comment (ghoshbishakh) [inline festflow/templates/festflow/keynote.html]: Keynote logo should be image. Your model does not have a logo.
keynote.image

File: festflow/templates/festflow/keynote.html [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    <img src="{{ keynote.logo.url }}" alt= "{{ keynote.name }}" />
+    {% endif %}
+    <h2>{{ keynote.name }}</h2>
+    <button>View</button>
```
Comment (ghoshbishakh) [inline festflow/templates/festflow/keynote.html]: No need of a button

File: festflow/templates/festflow/keynote_view.html [hunk 5]
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
+<html>
+<head>
+<title>{{ keynote.name }}</title>
+</head>
+<body>
+
+<h1>{{ keynote.name }}</h1>
+{% if keynote.logo %}
```
Comment (ghoshbishakh) [inline festflow/templates/festflow/keynote_view.html]: No model field as logo. That should be image I imgine.

File: festflow/views.py [hunk 6]
```diff
@@ -81,6 +87,21 @@ def event_view(request, event_identifier):
 
     return render(request, 'festflow/event_view.html', context)
 
+def keynote_view(request, keynote_identifier):
+    context = {}
+    try:
+        keynote = Keynote.objects.get(identifier=keynote_identifier)
+    except ObjectDoesNotExist:
+        raise Http404
```
Comment (nkprince007) [inline festflow/views.py]: You could use django's shortcut `get_object_or_get_404` here.

File: festflow/templates/festflow/keynote_view.html (hunk 2 above)
Comment (nkprince007) [inline festflow/templates/festflow/keynote_view.html]: And also no proper indentation.

File: festflow/admin.py [hunk 7]
```diff
@@ -35,6 +35,7 @@ class NewsLetterAdmin(admin.ModelAdmin):
 
 admin.site.register(Profile)
 admin.site.register(Event)
+admin.site.register(Keynote)
```
Comment (nkprince007) [inline festflow/admin.py]: :P Not an issue, but you could just add it at the end, right? this helps keeping git blame clear.

File: festflow/views.py (hunk 6 above)
Comment (ghoshbishakh) [inline festflow/views.py]: I think that is not necessary

**Comment by devanshgoenka97** [comment]
@ghoshbishakh  @nkprince007  The requested changes have been made. Kindly review

File: festflow/templates/festflow/keynote_view.html [hunk 8]
```diff
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
+<html>
+  <head>
+	<title>{{ keynote.name }}</title>
```
Comment (nkprince007) [inline festflow/templates/festflow/keynote_view.html]: follow something uniform in all `html` files, you're using 4 spaces here, and all other files follow 2 spaces

File: festflow/urls.py (hunk 1 above)
Comment (nkprince007) [inline festflow/urls.py]: Remove the spaces as noted by @ghoshbishakh

File: festflow/urls.py [hunk 9]
```diff
@@ -28,26 +31,24 @@
...
+    url(r'^login/$',views.login_page, name='login_page'),
 
 
     # social login urls
     url('', include('social.apps.django_app.urls', namespace='social')),
 
     # Form to complete profile
-    url(r'^complete_profile/$', views.complete_profile,
-        name='complete_profile'),
+    url(r'^complete_profile/$', views.complete_profile,name='complete_profile'),
```
Comment (nkprince007) [inline festflow/urls.py]: oh come on.. why are you changing the lines unnecessarily?

File: festflow/urls.py [hunk 10]
```diff
@@ -28,26 +31,24 @@
...
 
     # Event Register Page
-    url(r'^register_event/(?P<event_identifier>[a-z]*)/$',
-        views.register_event, name='register_event'),
+    url(r'^register_event/(?P<event_identifier>[a-z]*)/$',views.register_event, name='register_event'),
 
     # Login Page
-    url(r'^login/$',
-        views.login_page, name='login_page'),
+    url(r'^login/$',views.login_page, name='login_page'),
```
Comment (nkprince007) [inline festflow/urls.py]: this one too!?

File: festflow/urls.py [hunk 11]
```diff
@@ -28,26 +31,24 @@
...
-        name='event_view'),
+    url(r'^events/(?P<event_identifier>[a-z]*)/$', views.event_view,name='event_view'),
+
+    # Keynote View Page
+    url(r'^keynote/(?P<keynote_identifier>[a-z]*)/$', views.keynote_view,name='keynote_view'),
 
     # Event Register Page
-    url(r'^register_event/(?P<event_identifier>[a-z]*)/$',
-        views.register_event, name='register_event'),
+    url(r'^register_event/(?P<event_identifier>[a-z]*)/$',views.register_event, name='register_event'),
```
Comment (nkprince007) [inline festflow/urls.py]: and this too..

File: festflow/urls.py [hunk 12]
```diff
@@ -28,26 +31,24 @@
     url(r'^subscribe/$', views.subscribe, name='subscribe'),
 
     # Event View Page
-    url(r'^events/(?P<event_identifier>[a-z]*)/$', views.event_view,
-        name='event_view'),
+    url(r'^events/(?P<event_identifier>[a-z]*)/$', views.event_view,name='event_view'),
```
Comment (nkprince007) [inline festflow/urls.py]: Why did you have to change this?

File: festflow/urls.py [hunk 13]
```diff
@@ -28,26 +31,24 @@
...
 
     # Form to complete profile
-    url(r'^complete_profile/$', views.complete_profile,
-        name='complete_profile'),
+    url(r'^complete_profile/$', views.complete_profile,name='complete_profile'),
 
     # logout url
-    url(r'^logout/$', logout,
-        {'next_page': '/'})
+    url(r'^logout/$', logout,{'next_page': '/'})
```
Comment (nkprince007) [inline festflow/urls.py]: 😭  Please stop changing lines unnecessarily

**Comment by devanshgoenka97** [comment]
@nkprince007 There were unnecessary spaces present, I removed them and consolidated it into a single line

**Comment by nkprince007** [comment]
A space `logout,{` here is necessary, no need to remove that.

**Comment by ghoshbishakh** [comment]
@nkprince007 sorry I didnot realize he had made more changes which were not there earlier

**Comment by ghoshbishakh** [comment]
@nkprince007 He does not understand the meaning of unnecessary spaces. He didnot fix what I wanted him to. But I didnot notice that he had done some more changes

**Comment by ghoshbishakh** [comment]
@devanshgoenka97 please be careful. I'll have to be more careful from now on before merging

**Pull request merged**